### DISCUSSION TITLE
Where is write-message?

### DISCUSSION THREAD

**Discussion by MarkIannucci** [dataplat/dbatools#7037] [Q&A] [answered]
Hi, I'd like to read the current code behind write-message, but I can't find it anywhere.  I tried searching [write-message.ps1](https://github.com/sqlcollaborative/dbatools/search?q=write-message.ps1), and also looked for "[function write-message"](https://github.com/sqlcollaborative/dbatools/search?q=%22function+write-message%22&type=code), both come up short.  It seems at one point #2866 , it was in internal/Write-message.ps1, but I haven't found where it moved.

**Comment by wsmelton** [comment, marked as answer]
We have `Write-Message` written as a cmdlet so it is located within the `dbatools.dll`. You can find that in the VS sub-project of our repo [here](https://github.com/sqlcollaborative/dbatools/blob/cd7a897cf042666e3bdbd18b6fdefb77dd4991fd/bin/projects/dbatools/dbatools/Commands/WriteMessageCommand.cs).

It is loosely based on what is now `Write-PFSMessage` in @FriedrichWeinmann [PSFramework module](https://psframework.org/documentation/commands/PSFramework/Write-PSFMessage.html).
